managedtenants/utils/ocm.py

```diff
@@ -127,12 +146,31 @@ def upsert_addon(self, metadata):
 
         return addon
 
+    def _imageset_from_metadata(self, metadata):
+        imageset = {
+            "id": metadata.get("addonImageSetVersion"),
+            "enabled": metadata.get("enabled"),
+        }
+
+        for key, val in metadata.get("imageset").items():
+            if key in self.IMAGESET_KEYS:
+                imageset[self.IMAGESET_KEYS[key]] = val
+        return imageset
+
     def _addon_from_metadata(self, metadata):
         addon = {}
         metadata["addOnParameters"] = metadata.get("addOnParameters", [])
         metadata["addOnRequirements"] = metadata.get("addOnRequirements", [])
         for key, val in metadata.items():
             if key in self.ADDON_KEYS:
+                # Skip adding these parameters as they're present
+                # in the ImageSet (versions endpoint)
+                if metadata.get("addonImageSetVersion") and key in [
```

Comment:
nit: use set operations:

```python
if metadata.get("addonImageSetVersion") and key in (ADDON_KEYS - IMAGESET_KEYS):
```

Reply:
The intention of this condition is to ensure that the versioned fields are not populated into the addon endpoint payload (old endpoint) if the addon uses imagesets (because these fields will be posted against the new versions endpoint instead, so no need to populate them here anymore).

The only extra field in `IMAGESET_KEYS` is `indexImage` which will anyway always be absent if `addonImageSetVersion` is present. 

We could use `IMAGESET_KEYS` too, shouldn't make a difference. IMO I just wanted to be explicit about these 3 fields as they're the ones that should actually be skipped when using the new endpoint. `indexImage` will simply never be present in the case where we're using imageSets.